table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +97,537 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
+     * Returns the list of tables.
+     * @return list of tables
      */
-    public Set<BypassTableEntry> getBypassTableEntries() {
-      return mEntries;
+    public IncludeExcludeList<T> getList() {
+      return mList;
     }
   }
 
   /**
-   * Table to partitions mapping.
+   * Deserialize to an IncludeExcludeList of given inner type T.
+   * When the contained object is an array, an include list is implied.
+   * @param <T> the contained inner entry type
    */
-  @JsonDeserialize(using = BypassTableEntryDeserializer.class)
-  public static class BypassTableEntry {
-    private final String mTableName;
-    private final Set<String> mPartitions;
+  static class TablesEntryDeserializer<T extends NamePatternEntry> {
+    IncludeExcludeList<T> deserializeToList(
+        Class<T> type, JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+      JsonNode node = mapper.readTree(jp);
+      if (node == null) {
+        return null;
+      }
+      if (!node.hasNonNull("tables")) {
+        throw new JsonParseException(mapper.treeAsTokens(node), "missing field `tables`");
+      }
+      node = node.get("tables");
+      if (node.isArray()) {
+        // in case of an array, an included list is implied
+        Set<T> entries = mapper.convertValue(
+            node,
+            mapper.getTypeFactory().constructCollectionType(Set.class, type)
+        );
+        return new IncludeExcludeList<>(entries);
+      }
+      if (node.isObject()) {
+        // otherwise, deserialize as an IncludeExcludeList object
+        return mapper.convertValue(
+            node,
+            mapper.getTypeFactory().constructParametricType(IncludeExcludeList.class, type)
+        );
+      }
+      throw new JsonParseException(mapper.treeAsTokens(node),
+          "invalid syntax, expecting array or object");
+    }
+  }
+
+  static class IgnoreTablesSpecDeserializer extends JsonDeserializer<IgnoreTablesSpec> {
+    @Override
+    public IgnoreTablesSpec deserialize(JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      TablesEntryDeserializer<NamePatternEntry> deserializer = new TablesEntryDeserializer<>();
+      IncludeExcludeList<NamePatternEntry> list =
+          deserializer.deserializeToList(NamePatternEntry.class, jp, cxt);
+      return new IgnoreTablesSpec(list);
+    }
+  }
+
+  static class BypassTablesSpecDeserializer extends JsonDeserializer<BypassTablesSpec> {
+    @Override
+    public BypassTablesSpec deserialize(JsonParser jp, DeserializationContext cxt)
+        throws IOException, JsonProcessingException {
+      TablesEntryDeserializer<TableEntry> deserializer = new TablesEntryDeserializer<>();
+      IncludeExcludeList<TableEntry> list =
+          deserializer.deserializeToList(TableEntry.class, jp, cxt);
+      return new BypassTablesSpec(list);
+    }
+  }
```

Comment:
These two deserializers differ only in their type arguments: ignored tables does not support partition specification, so the inner type is `NamePatternEntry`; bypassed tables do, so the inner type is `TableEntry`, which is a subclass of `NamePatternEntry`, and supports partition specification.